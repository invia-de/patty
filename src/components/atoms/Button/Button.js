import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../utils/classnames';

import styles from './Button.module.scss';

export default function Button({
  children,
  className,
  onClick,
  secondary,
  block,
  ...props
}) {
  return (
    <button
      className={cx(
        styles.button,
        secondary ? styles.secondary : null,
        block ? styles.block : null,
        className
      )}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** required due to accessibility */
  children: PropTypes.node.isRequired,
  /** What should happen when the button is pressed */
  onClick: PropTypes.func.isRequired,
  /** should the secondary style be rendered */
  secondary: PropTypes.bool,
  /** Show button as a 100% wide block */
  block: PropTypes.bool
};
