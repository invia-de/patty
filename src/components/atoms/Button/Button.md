```jsx
<Button onClick={() => alert('primary button clicked')}>Primary Button</Button>

<Button secondary onClick={() => alert('secondary button clicked')}>Secondary Button</Button>
```

Block button:

```jsx
<Button secondary block onClick={() => alert('secondary block button clicked')}>
  Secondary Block Button
</Button>
```
