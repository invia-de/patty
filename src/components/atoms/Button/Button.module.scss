@import 'theme.scss';

$button-primary-text-color: $white !default;
$button-primary-bg-color: $blue !default;
$button-secondary-text-color: $blue !default;
$button-secondary-bg-color: $blue-bright !default;

.button,
.button:active,
.button:hover,
.button:focus {
  padding: 8px 14px;
  line-height: 16px;
  font-size: 14px;
  font-weight: bold;
  color: $button-primary-text-color;
  background: $button-primary-bg-color;
  border: none;
  outline: 0;
}

.secondary,
.secondary:active,
.secondary:hover,
.secondary:focus {
  color: $button-secondary-text-color;
  background: $button-secondary-bg-color;
}

.block {
  text-align: center;
  width: 100%;
  display: block;
}
