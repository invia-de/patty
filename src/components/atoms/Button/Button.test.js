import React from 'react';
import Button from './Button';
import { render } from '@testing-library/react';

test('Button renders and clicks', () => {
  const onClick = jest.fn();
  const { container } = render(<Button onClick={onClick} />);

  expect(container.firstChild).toMatchSnapshot();
  container.firstChild.click();
  expect(onClick).toHaveBeenCalled();
});
