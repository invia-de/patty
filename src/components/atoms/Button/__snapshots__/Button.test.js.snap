// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders and clicks 1`] = `
<button
  class="button"
/>
`;
